app/assets/javascripts/discourse/tests/integration/components/bookmark-test.js

```diff
@@ -0,0 +1,155 @@
+
+  hooks.beforeEach(function () {
+    KeyboardShortcutInitializer.initialize(getApplication());
+    this.actions.registerOnCloseHandler = () => {};
+    this.actions.closeModal = () => {};
+    this.setProperties({
+      model: {},
+      afterSave: () => {},
+      afterDelete: () => {},
+      onCloseWithoutSaving: () => {},
+    });
+  });
+
+  hooks.afterEach(function () {
+    if (clock) {
+      clock.restore();
+    }
+    sinon.restore();
+  });
+
+  componentTest("show later this week option if today is < Thursday", {
+    template: template,
+
+    beforeEach() {
+      mockMomentTz("2019-12-10T08:00:00", this.currentUser._timezone);
+    },
+
+    test(assert) {
+      assert.ok(exists("#tap_tile_later_this_week"), "it has later this week");
+    },
+  });
+
+  componentTest(
+    "does not show later this week option if today is >= Thursday",
+    {
+      template: template,
+
+      beforeEach() {
+        mockMomentTz("2019-12-13T08:00:00", this.currentUser._timezone);
+      },
+
+      test(assert) {
+        assert.notOk(
+          exists("#tap_tile_later_this_week"),
+          "it does not have later this week"
+        );
+      },
+    }
+  );
+
+  componentTest("later today does not show if later today is tomorrow", {
+    template: template,
+
+    beforeEach() {
+      mockMomentTz("2019-12-11T22:00:00", this.currentUser._timezone);
+    },
+
+    test(assert) {
+      assert.notOk(
+        exists("#tap_tile_later_today"),
+        "it does not have later today"
+      );
+    },
+  });
+
+  componentTest("later today shows if it is after 5pm but before 6pm", {
+    template: template,
+
+    beforeEach() {
+      mockMomentTz("2019-12-11T14:30:00", this.currentUser._timezone);
+    },
+
+    test(assert) {
+      assert.ok(exists("#tap_tile_later_today"), "it does have later today");
+    },
+  });
+
+  componentTest("later today does not show if it is after 5pm", {
+    template: template,
+
+    beforeEach() {
+      mockMomentTz("2019-12-11T17:00:00", this.currentUser._timezone);
+    },
+
+    test(assert) {
+      assert.notOk(
+        exists("#tap_tile_later_today"),
+        "it does not have later today"
+      );
+    },
+  });
+
+  componentTest("later today does show if it is before the end of the day", {
+    template: template,
+
+    beforeEach() {
+      mockMomentTz("2019-12-11T13:00:00", this.currentUser._timezone);
+    },
+
+    test(assert) {
+      assert.ok(exists("#tap_tile_later_today"), "it does have later today");
+    },
+  });
+
+  componentTest("prefills the custom reminder type date and time", {
+    template: template,
+
+    beforeEach() {
+      let name = "test";
+      let reminderAt = "2020-05-15T09:45:00";
+      this.model = { id: 1, name: name, reminderAt: reminderAt };
+    },
+
+    test(assert) {
+      assert.equal(queryAll("#bookmark-name")[0].value, "test");
+      assert.equal(
+        queryAll("#custom-date > .date-picker")[0].value,
```

Comment:
just use query if you want only the first one and not queryAll (you can import it too)